base/server/upgrade/10.8.2/01-FixECAdminCertProfile.py

```diff
@@ -30,6 +30,24 @@ def upgrade_subsystem(self, instance, subsystem):
         path = subsystem.config.get('profile.caECAdminCert.config')
         logger.info('Current path: %s', path)
 
+        if path is None:
+            # The attribute is missing. Patch the missing attribute
+            logger.info('profile.caECAdminCert.config missing in CS.cfg')
+
+            path = "{0}/profiles/{1}/caAdminCert.cfg".format(
```

Comment:
I think we should add the missing path with the correct value directly, and only fix the path if it already exists (presumably with the wrong value) like this:
```
path = subsystem.config['profile.caECAdminCert.config']
if path is None:
    # add missing path
    path = "{0}/profiles/{1}/caECAdminCert.cfg".format(
        subsystem.base_dir, subsystem.name)

else:
    # fix existing path
    dirname = os.path.dirname(path)
    path = os.path.join(dirname, 'caECAdminCert.cfg')

subsystem.config['profile.caECAdminCert.config'] = path
```

Reply:
On the second thought, I think we can simply set the correct path regardless of the existing value:
```
path = "{0}/profiles/{1}/caECAdminCert.cfg".format(
        subsystem.base_dir, subsystem.name)
subsystem.config['profile.caECAdminCert.config'] = path
```
But the current code will also work just fine.